### PR TITLE
Remix upgrade

### PR DESCRIPTION
**Pull request by smatthies**

### CODE REVIEW

File: netlify.toml [hunk 1]
```diff
@@ -1,17 +1,16 @@
 [build]
-  command = "remix build"
-  publish = "public"
+command = "npm run build"
+publish = "build/client"
```
Comment (HerrBertling) [inline netlify.toml:3]: I'm not sure if this is correct? In the pipeline log (I think it is publicly accessible? => https://app.netlify.com/sites/keen-liskov-77de74/deploys/66eb34069a3c1f0008085b6b ), there's only the client part mentioned in the deployment step. But we definitely also need the server part. That should be the reason why you cannot access the page.

File: vite.config.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { vitePlugin as remix } from "@remix-run/dev";
+import { defineConfig } from "vite";
+import tsconfigPaths from "vite-tsconfig-paths";
+import { netlifyPlugin } from "@netlify/remix-edge-adapter/plugin";
+import { remixDevTools } from "remix-development-tools";
+
+
+export default defineConfig({
```
Comment (HerrBertling) [inline vite.config.ts:8]: I think you need to add the `netlifyPlugin` to the `plugins` array, it is not mentioned in there right now.

File: netlify.toml (hunk 1 above)
Comment (smatthies) [reply netlify.toml:3]: Yeah, but I had the same thought as well and checked it that to be sure :D

**Comment by HerrBertling** [comment]
@smatthies I've thrown a few commits into this and now it looks like the deployment is working :) https://deploy-preview-275--keen-liskov-77de74.netlify.app/de

File: package.json [hunk 3]
```diff
@@ -1,55 +1,68 @@
...
+    "eslint-plugin-react-hooks": "^4.6.0",
+    "postcss": "^8.4.39",
+    "prettier": "^3.3.3",
+    "typescript": "^5.1.6",
+    "vite": "^5.0.0",
+    "vite-tsconfig-paths": "^4.2.1"
   },
   "engines": {
-    "node": ">=18"
+    "node": ">=22.2.0"
```
Comment (smatthies) [inline package.json:66]: Wieso hab ich das denn übersehen 🙃 zu viel mit den Versionen rumgespielt..

**Comment by HerrBertling** [comment]
I‘m fine with merging this whenever there’s time for a sanity check and eventual rollback (which I assume will not be needed). So whenever you’re ready! I‘m also happy with hitting that button myself 😅

**Pull request merged**